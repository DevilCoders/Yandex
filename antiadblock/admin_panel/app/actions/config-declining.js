export const OPEN_CONFIG_DECLINING = 'OPEN_CONFIG_DECLINING';
export const CLOSE_CONFIG_DECLINING = 'CLOSE_CONFIG_DECLINING';

export function openConfigDeclining(serviceId, labelId, id) {
    return {
        type: OPEN_CONFIG_DECLINING,
        serviceId,
        labelId,
        id
    };
}

export function closeConfigDeclining(id) {
    return {
        type: CLOSE_CONFIG_DECLINING,
        id
    };
}

