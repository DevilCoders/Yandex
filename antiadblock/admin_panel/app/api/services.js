import {antiadbRequest} from 'app/lib/api';

export default {

    fetchServices: function() {
        return antiadbRequest({
            path: '/services',
            options: {
                method: 'GET'
            }
        });
    },

    searchConfigs: function(pattern, offset, limit, active) {
        return antiadbRequest({
            path: '/search',
            options: {
                method: 'GET'
            },
            queryParams: {
                pattern,
                offset,
                limit,
                active
            }
        });
    },

    fetchAlerts: function() {
        return antiadbRequest({
            path: '/get_all_checks',
            options: {
                method: 'GET'
            }
        });
    }
};
