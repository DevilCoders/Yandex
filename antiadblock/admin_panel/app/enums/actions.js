export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
