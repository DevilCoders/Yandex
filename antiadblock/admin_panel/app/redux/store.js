import {createStore, compose, applyMiddleware} from 'redux';
import reducer from 'app/reducers/index';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';

let composeEnhancers = compose;

export function configureStore(state = {}) {
    if (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__) {
        composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
    }
    if (process.env.NODE_ENV === 'local') {
        return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware, createLogger())));
    }
    return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware)));
}
