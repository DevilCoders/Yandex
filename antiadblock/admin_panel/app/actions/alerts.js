import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
