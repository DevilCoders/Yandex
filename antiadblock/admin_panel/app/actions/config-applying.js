import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
