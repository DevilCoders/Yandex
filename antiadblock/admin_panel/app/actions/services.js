import servicesApi from 'app/api/services';
import {setGlobalErrors} from 'app/actions/errors';

export const START_SERVICES_LOADING = 'START_SERVICES_LOADING';
export const END_SERVICES_LOADING = 'END_SERVICES_LOADING';
export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchServices() {
    return dispatch => {
        dispatch(startServicesLoading());

        return servicesApi.fetchServices().then(services => {
            return dispatch(endServicesLoading(services.items));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startServicesLoading() {
    return {
        type: START_SERVICES_LOADING
    };
}

export function endServicesLoading(items) {
    return {
        type: END_SERVICES_LOADING,
        items
    };
}
