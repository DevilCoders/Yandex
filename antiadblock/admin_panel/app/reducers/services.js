import * as servicesActions from 'app/actions/services';
import update from 'app/lib/update';
import * as serviceActions from 'app/actions/service';
import {STATUS} from 'app/enums/service';

export default function(state, action) {
    const {type} = action;

    state = state || {
        items: [],
        loading: false,
        loaded: false
    };

    switch (type) {
        case servicesActions.START_SERVICES_LOADING: {
            return update(state, {
                loading: {
                    $set: true
                }
            });
        }
        case servicesActions.END_SERVICES_LOADING: {
            return update(state, {
                $merge: {
                    items: action.items,
                    loading: false,
                    loaded: true
                }
            });
        }
        case serviceActions.END_SERVICE_DISABLING: {
            return update(state, {
                $byId: {
                    path: 'items',
                    key: 'id',
                    id: action.serviceId,
                    fallback: {},
                    do: {
                        $merge: {
                            status: STATUS.INACTIVE
                        }
                    }
                }
            });
        }
        case serviceActions.END_SERVICE_ENABLING: {
            return update(state, {
                $byId: {
                    path: 'items',
                    key: 'id',
                    id: action.serviceId,
                    fallback: {},
                    do: {
                        $merge: {
                            status: STATUS.ACTIVE
                        }
                    }
                }
            });
        }
        default: {
            return state;
        }
    }
}
