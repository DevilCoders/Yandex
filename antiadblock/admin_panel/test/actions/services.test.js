import * as servicesActions from 'app/actions/services';
import * as servicesActionsFixtures from '../fixtures/actions/services';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as storeFixtures from '../fixtures/store';
import {SERVICES} from '../fixtures/backend/services';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('services actions', () => {
    it('should create START_SERVICES_LOADING', () => {
        expect(servicesActions.startServicesLoading()).toEqual(servicesActionsFixtures.START_SERVICES_LOADING);
    });

    it('should create END_SERVICES_LOADING', () => {
        expect(servicesActions.endServicesLoading(SERVICES.items)).toEqual(servicesActionsFixtures.END_SERVICES_LOADING);
    });

    it('should create END_SERVICES_LOADING after loading', () => {
        const store = mockStore(storeFixtures.SERVICES_EMPTY);

        fetch.mockResponseOnce(JSON.stringify(SERVICES));

        return store.dispatch(servicesActions.fetchServices(123, 0, 20)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create END_SERVICES_LOADING AND SET_ERRORS', () => {
        const store = mockStore(storeFixtures.SERVICES_EMPTY);

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(servicesActions.fetchServices(123, 0, 20)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });
});
