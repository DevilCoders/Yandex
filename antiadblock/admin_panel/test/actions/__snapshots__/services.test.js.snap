// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`services actions should create END_SERVICES_LOADING AND SET_ERRORS 1`] = `
Array [
  Object {
    "type": "START_SERVICES_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`services actions should create END_SERVICES_LOADING after loading 1`] = `
Array [
  Object {
    "type": "START_SERVICES_LOADING",
  },
  Object {
    "items": Array [
      Object {
        "id": 1,
        "name": "Test1",
        "status": "ok",
      },
      Object {
        "id": 2,
        "name": "TestTest2",
        "status": "inactive",
      },
      Object {
        "id": 3,
        "name": "TestTestTest3",
        "status": "inactive",
      },
      Object {
        "id": 4,
        "name": "TestTestTestTest4",
        "status": "inactive",
      },
      Object {
        "id": 5,
        "name": "TestTestTestTestTest5",
        "status": "inactive",
      },
      Object {
        "id": 6,
        "name": "TestTestTestTestTestTest6",
        "status": "inactive",
      },
      Object {
        "id": 7,
        "name": "TestTestTestTestTestTestTest7",
        "status": "inactive",
      },
      Object {
        "id": 8,
        "name": "TestTestTestTestTestTestTestTest8",
        "status": "inactive",
      },
      Object {
        "id": 9,
        "name": "TestTestTestTestTestTestTestTestTest9",
        "status": "inactive",
      },
      Object {
        "id": 10,
        "name": "TestTestTestTestTestTestTestTestTestTest10",
        "status": "inactive",
      },
      Object {
        "id": 11,
        "name": "Test11",
        "status": "inactive",
      },
      Object {
        "id": 12,
        "name": "TestTest12",
        "status": "inactive",
      },
      Object {
        "id": 13,
        "name": "TestTestTest13",
        "status": "inactive",
      },
      Object {
        "id": 14,
        "name": "TestTestTestTest14",
        "status": "inactive",
      },
      Object {
        "id": 15,
        "name": "TestTestTestTestTest15",
        "status": "inactive",
      },
      Object {
        "id": 16,
        "name": "TestTestTestTestTestTest16",
        "status": "inactive",
      },
      Object {
        "id": 17,
        "name": "TestTestTestTestTestTestTest17",
        "status": "inactive",
      },
      Object {
        "id": 18,
        "name": "TestTestTestTestTestTestTestTest18",
        "status": "inactive",
      },
      Object {
        "id": 19,
        "name": "TestTestTestTestTestTestTestTestTest19",
        "status": "inactive",
      },
      Object {
        "id": 20,
        "name": "TestTestTestTestTestTestTestTestTestTest20",
        "status": "inactive",
      },
      Object {
        "id": 21,
        "name": "Test21",
        "status": "inactive",
      },
      Object {
        "id": 22,
        "name": "TestTest22",
        "status": "inactive",
      },
      Object {
        "id": 23,
        "name": "TestTestTest23",
        "status": "inactive",
      },
      Object {
        "id": 24,
        "name": "TestTestTestTest24",
        "status": "inactive",
      },
      Object {
        "id": 25,
        "name": "TestTestTestTestTest25",
        "status": "inactive",
      },
      Object {
        "id": 26,
        "name": "TestTestTestTestTestTest26",
        "status": "inactive",
      },
      Object {
        "id": 27,
        "name": "TestTestTestTestTestTestTest27",
        "status": "inactive",
      },
      Object {
        "id": 28,
        "name": "TestTestTestTestTestTestTestTest28",
        "status": "inactive",
      },
      Object {
        "id": 29,
        "name": "TestTestTestTestTestTestTestTestTest29",
        "status": "inactive",
      },
      Object {
        "id": 30,
        "name": "TestTestTestTestTestTestTestTestTestTest30",
        "status": "inactive",
      },
      Object {
        "id": 31,
        "name": "Test31",
        "status": "inactive",
      },
      Object {
        "id": 32,
        "name": "TestTest32",
        "status": "inactive",
      },
      Object {
        "id": 33,
        "name": "TestTestTest33",
        "status": "inactive",
      },
      Object {
        "id": 34,
        "name": "TestTestTestTest34",
        "status": "inactive",
      },
      Object {
        "id": 35,
        "name": "TestTestTestTestTest35",
        "status": "inactive",
      },
      Object {
        "id": 36,
        "name": "TestTestTestTestTestTest36",
        "status": "inactive",
      },
      Object {
        "id": 37,
        "name": "TestTestTestTestTestTestTest37",
        "status": "inactive",
      },
      Object {
        "id": 38,
        "name": "TestTestTestTestTestTestTestTest38",
        "status": "inactive",
      },
      Object {
        "id": 39,
        "name": "TestTestTestTestTestTestTestTestTest39",
        "status": "inactive",
      },
      Object {
        "id": 40,
        "name": "TestTestTestTestTestTestTestTestTestTest40",
        "status": "inactive",
      },
      Object {
        "id": 41,
        "name": "Test41",
        "status": "inactive",
      },
      Object {
        "id": 42,
        "name": "TestTest42",
        "status": "inactive",
      },
      Object {
        "id": 43,
        "name": "TestTestTest43",
        "status": "inactive",
      },
      Object {
        "id": 44,
        "name": "TestTestTestTest44",
        "status": "inactive",
      },
      Object {
        "id": 45,
        "name": "TestTestTestTestTest45",
        "status": "inactive",
      },
      Object {
        "id": 46,
        "name": "TestTestTestTestTestTest46",
        "status": "inactive",
      },
      Object {
        "id": 47,
        "name": "TestTestTestTestTestTestTest47",
        "status": "inactive",
      },
      Object {
        "id": 48,
        "name": "TestTestTestTestTestTestTestTest48",
        "status": "inactive",
      },
      Object {
        "id": 49,
        "name": "TestTestTestTestTestTestTestTestTest49",
        "status": "inactive",
      },
      Object {
        "id": 50,
        "name": "TestTestTestTestTestTestTestTestTestTest50",
        "status": "inactive",
      },
      Object {
        "id": 51,
        "name": "Test51",
        "status": "inactive",
      },
      Object {
        "id": 52,
        "name": "TestTest52",
        "status": "inactive",
      },
      Object {
        "id": 53,
        "name": "TestTestTest53",
        "status": "inactive",
      },
      Object {
        "id": 54,
        "name": "TestTestTestTest54",
        "status": "inactive",
      },
      Object {
        "id": 55,
        "name": "TestTestTestTestTest55",
        "status": "inactive",
      },
      Object {
        "id": 56,
        "name": "TestTestTestTestTestTest56",
        "status": "inactive",
      },
      Object {
        "id": 57,
        "name": "TestTestTestTestTestTestTest57",
        "status": "inactive",
      },
      Object {
        "id": 58,
        "name": "TestTestTestTestTestTestTestTest58",
        "status": "inactive",
      },
      Object {
        "id": 59,
        "name": "TestTestTestTestTestTestTestTestTest59",
        "status": "inactive",
      },
      Object {
        "id": 60,
        "name": "TestTestTestTestTestTestTestTestTestTest60",
        "status": "inactive",
      },
      Object {
        "id": 61,
        "name": "Test61",
        "status": "inactive",
      },
      Object {
        "id": 62,
        "name": "TestTest62",
        "status": "inactive",
      },
      Object {
        "id": 63,
        "name": "TestTestTest63",
        "status": "inactive",
      },
      Object {
        "id": 64,
        "name": "TestTestTestTest64",
        "status": "inactive",
      },
      Object {
        "id": 65,
        "name": "TestTestTestTestTest65",
        "status": "inactive",
      },
      Object {
        "id": 66,
        "name": "TestTestTestTestTestTest66",
        "status": "inactive",
      },
      Object {
        "id": 67,
        "name": "TestTestTestTestTestTestTest67",
        "status": "inactive",
      },
      Object {
        "id": 68,
        "name": "TestTestTestTestTestTestTestTest68",
        "status": "inactive",
      },
      Object {
        "id": 69,
        "name": "TestTestTestTestTestTestTestTestTest69",
        "status": "inactive",
      },
      Object {
        "id": 70,
        "name": "TestTestTestTestTestTestTestTestTestTest70",
        "status": "inactive",
      },
      Object {
        "id": 71,
        "name": "Test71",
        "status": "inactive",
      },
      Object {
        "id": 72,
        "name": "TestTest72",
        "status": "inactive",
      },
      Object {
        "id": 73,
        "name": "TestTestTest73",
        "status": "inactive",
      },
      Object {
        "id": 74,
        "name": "TestTestTestTest74",
        "status": "inactive",
      },
      Object {
        "id": 75,
        "name": "TestTestTestTestTest75",
        "status": "inactive",
      },
      Object {
        "id": 76,
        "name": "TestTestTestTestTestTest76",
        "status": "inactive",
      },
      Object {
        "id": 77,
        "name": "TestTestTestTestTestTestTest77",
        "status": "inactive",
      },
      Object {
        "id": 78,
        "name": "TestTestTestTestTestTestTestTest78",
        "status": "inactive",
      },
      Object {
        "id": 79,
        "name": "TestTestTestTestTestTestTestTestTest79",
        "status": "inactive",
      },
      Object {
        "id": 80,
        "name": "TestTestTestTestTestTestTestTestTestTest80",
        "status": "inactive",
      },
      Object {
        "id": 81,
        "name": "Test81",
        "status": "inactive",
      },
      Object {
        "id": 82,
        "name": "TestTest82",
        "status": "inactive",
      },
      Object {
        "id": 83,
        "name": "TestTestTest83",
        "status": "inactive",
      },
      Object {
        "id": 84,
        "name": "TestTestTestTest84",
        "status": "inactive",
      },
      Object {
        "id": 85,
        "name": "TestTestTestTestTest85",
        "status": "inactive",
      },
      Object {
        "id": 86,
        "name": "TestTestTestTestTestTest86",
        "status": "inactive",
      },
      Object {
        "id": 87,
        "name": "TestTestTestTestTestTestTest87",
        "status": "inactive",
      },
      Object {
        "id": 88,
        "name": "TestTestTestTestTestTestTestTest88",
        "status": "inactive",
      },
      Object {
        "id": 89,
        "name": "TestTestTestTestTestTestTestTestTest89",
        "status": "inactive",
      },
      Object {
        "id": 90,
        "name": "TestTestTestTestTestTestTestTestTestTest90",
        "status": "inactive",
      },
      Object {
        "id": 91,
        "name": "Test91",
        "status": "inactive",
      },
      Object {
        "id": 92,
        "name": "TestTest92",
        "status": "inactive",
      },
      Object {
        "id": 93,
        "name": "TestTestTest93",
        "status": "inactive",
      },
      Object {
        "id": 94,
        "name": "TestTestTestTest94",
        "status": "inactive",
      },
      Object {
        "id": 95,
        "name": "TestTestTestTestTest95",
        "status": "inactive",
      },
      Object {
        "id": 96,
        "name": "TestTestTestTestTestTest96",
        "status": "inactive",
      },
      Object {
        "id": 97,
        "name": "TestTestTestTestTestTestTest97",
        "status": "inactive",
      },
      Object {
        "id": 98,
        "name": "TestTestTestTestTestTestTestTest98",
        "status": "inactive",
      },
      Object {
        "id": 99,
        "name": "TestTestTestTestTestTestTestTestTest99",
        "status": "inactive",
      },
      Object {
        "id": 100,
        "name": "TestTestTestTestTestTestTestTestTestTest100",
        "status": "inactive",
      },
    ],
    "type": "END_SERVICES_LOADING",
  },
]
`;
